Отправляем электронное письмо из сценария

M2rk13 opened the pull request:
---
:mortar_board: [Отправляем электронное письмо из сценария](https://up.htmlacademy.ru/php/12/user/1415187/tasks/18)

M2rk13 committed:
last task

00dev commented on functions.php:
```diff
@@ -988,13 +1101,231 @@ function search_posts(mysqli $connection, string $keywords)
+    if (!$settings['encryption']) {
+        $transport = new Swift_SmtpTransport($settings['server'], $settings['port']);
+    } else {
+        $transport = new Swift_SmtpTransport($settings['server'], $settings['port'], $setting['encryption']);
```
Здесь опечатка $setting → $settings

M2rk13 replied:
Окей, не заметил, спасибо

00dev commented on functions.php:
```diff
@@ -988,13 +1101,231 @@ function search_posts(mysqli $connection, string $keywords)
+{
+    $subject = 'У вас новый подписчик';
+    $message = new Swift_Message($subject);
+    $message->setFrom($sender, $site_name);
```
$site_name не определена в данной функции (нужно определить ее как глобальную)

M2rk13 replied:
Эта переменная уже содержится в подключаемом везде settings.php (через base.php), может ее лучше просто передать в функцию?

00dev reviewed (commented):
Общее замечание: следует всегда определять active_section для шаблона layout.php. Дополнительно следует просмотреть сайт и глянуть в лог, не появляются-ли сообщения вида
> PHP Notice:  Undefined variable: active_section in /home/froloff/project/my/htmlacademy/students/MarkMorgunov/1415187-readme-12/templates/layout.php on li
ne 106, referer: http://1415187-readme-12.devel/post.php?id=2

M2rk13 committed:
Внесены правки

Удален файл, демонстрирующий верстку страницы сообщений
Исправлена валидация длины сообщения (длина поля, при ошибке валидации текст сохраняется)
Исправлены опечатки в functions.php

M2rk13 committed:
-

M2rk13 committed:
-

M2rk13 committed:
Правки по Notice

Внесены правки по Notice в логах
Улучшено направление сортировки popular.php при переходе с другой страницы.
Улучшена валидация комментария к посту: теперь текст не пропадает при ошибке валидации

00dev commented on messages.php:
```diff
@@ -0,0 +1,65 @@
+}
+
+$dialogs = get_dialogs($connection, $user['id']);
+if (($dialogs !== null) || ($GET['id'] !== null)) {
```
Опечатка $GET → $_GET

00dev commented on functions.php:
```diff
@@ -402,17 +408,17 @@ function validate_exists(array $validation_array, string $parameter_name, array
     $sql = "SELECT $password_column_name FROM $table_name WHERE $users_column_name = ?";
-    $db_password = secure_query_bind_result($db_connection, $sql, false, $email);
+    $db_password = secure_query_bind_result($connection, $sql, false, $email);
     $password = mysqli_fetch_all($db_password, MYSQLI_ASSOC)[0]['password'];
```
Не всегда существует нулевой элемент. Если пользователь ошибается с адресом почты, то в результате будет 0 строк. Нужно проверять что вернлся непустой результат

00dev commented on functions.php:
```diff
@@ -988,13 +1107,234 @@ function search_posts(mysqli $connection, string $keywords)
+        ON users.id = dialog
+        ORDER BY last_message DESC ";
+    $dialogs_mysqli = secure_query_bind_result($connection, $select_dialogs_query, false, $user_id, $user_id, $user_id);
+    while ($dialogs = mysqli_fetch_array($dialogs_mysqli, MYSQLI_ASSOC)) {
```
Здесь нужно инициализировать `$dialogs_assoc` 

```
$dialogs_assoc = [];
```

Если результат запроса будет пустым, в строке 1153 будет предупреждение об обращении к несуществующей переменной

00dev commented on messages.php:
```diff
@@ -0,0 +1,65 @@
+        header("Location: " . $_SERVER['PHP_SELF']);
+    }
+}
+
```
Здесь нужно инициализировать `$active_dialog_id`. Если следующий блок if не выполнится - переменная не будет определена (дальше по коду она используется)

M2rk13 committed:
Update messages.php

M2rk13 committed:
update funtions.php & messages.php

M2rk13 committed:
-

M2rk13 committed:
Исправления

Доработка messsages.php:
-нельзя отправить себе сообщение, если явно задать id
-добавляется статус прочтения сообщения при его создании ('0')
-реализована возможность написать пользователю из профиля с отображением получателя в messages.php
- при отсутствии сообщений выведется соответствующая информация
-экранированы слабые места в шаблонах к add.php
-value теперь имеют кавычки, как и положено в разметке на шаблонах для add.php
-изменен почтовый сервер в mail.php
